mne_bids/report.py

```diff
@@ -0,0 +1,552 @@
+
+    # keep track of channel type, status
+    ch_status_count = {'bad': [], 'good': []}
+    ch_count = []
+
+    # loop through each scan
+    for scan_fpath in scans_fpaths:
+        # load in the scans.tsv file
+        # and read metadata for each scan
+        scans_tsv = _from_tsv(scan_fpath)
+        scans = scans_tsv['filename']
+        for scan in scans:
+            # summarize metadata of recordings
+            bids_basename, _ = _parse_ext(scan)
+            kind = op.dirname(scan)
+            if kind not in ['meg', 'eeg', 'ieeg']:
+                continue
+
+            # convert to BIDS Path
+            params = _parse_bids_filename(bids_basename, verbose)
+            bids_basename = BIDSPath(subject=params.get('sub'),
+                                     session=params.get('ses'),
+                                     recording=params.get('rec'),
+                                     acquisition=params.get('acq'),
+                                     processing=params.get('proc'),
+                                     space=params.get('space'),
+                                     run=params.get('run'),
+                                     task=params.get('task'),
+                                     prefix=bids_root)
+
+            # XXX: improve to allow emptyroom
+            if bids_basename.subject == 'emptyroom':
+                continue
+
+            channels_fname = _find_matching_sidecar(bids_fname=bids_basename,
+                                                    bids_root=bids_root,
+                                                    suffix='channels.tsv')
+
+            # summarize channels.tsv
+            channels_tsv = _from_tsv(channels_fname)
+            for status in ch_status_count.keys():
+                ch_status = [ch for ch in channels_tsv['status']
+                             if ch == status]
+                ch_status_count[status].append(len(ch_status))
+            ch_count.append(len(channels_tsv['name']))
+
+    # create summary template strings for status
+    template_dict = {
+        'mean_chs': np.mean(ch_count),
+        'std_chs': np.std(ch_count),
+        'mean_good_chs': np.mean(ch_status_count['good']),
+        'std_good_chs': np.std(ch_status_count['good']),
+        'mean_bad_chs': np.mean(ch_status_count['bad']),
+        'std_bad_chs': np.std(ch_status_count['bad']),
+    }
+    for key, val in template_dict.items():
+        template_dict[key] = round(val, 2)
+    return template_dict
+
+
+def make_report(bids_root, session=None, verbose=True):
+    """Create a methods paragraph string from BIDS dataset.
+
+    Summarizes the REQUIRED components in the BIDS specification
+    and also some RECOMMENDED components. Currently, the methods
+    paragraph summarize the:
+
+      - dataset_description.json file
+      - (optional) participants.tsv file
+      - (optional) modality-agnostic files for (M/I)EEG data,
+        which reads files from the *_scans.tsv file.
```

Comment:
I think this is the culprit for the red CIs. My suggestion should fix it. @adam2392 

```suggestion
        which reads files from the ``*_scans.tsv`` file.
```